pkg/console/subresource/configmap/configmap.go

```diff
@@ -40,54 +36,62 @@ func getApiUrl(infrastructureConfig *configv1.Infrastructure) string {
 	return ""
 }
 
-// DefaultConfigMap returns
-// - a new configmap,
-// - a bool indicating if config was merged (unsupportedConfigOverrides)
-// - an error
-func DefaultConfigMap(operatorConfig *operatorv1.Console, consoleConfig *configv1.Console, managedConfig *corev1.ConfigMap, infrastructureConfig *configv1.Infrastructure, rt *routev1.Route) (*corev1.ConfigMap, bool, error) {
-
-	// Build a default config that can be overwritten from other sources
-	host := rt.Spec.Host
-	apiServerURL := getApiUrl(infrastructureConfig)
-	defaultProviders := operatorv1.ConsoleProviders{}
-	defaultConfig := NewYamlConfig(host, defaultLogoutURL, DEFAULT_BRAND, DEFAULT_DOC_URL, apiServerURL, defaultProviders)
-	// Get console config from the openshift-config-managed namespace
+func statusPageId(operatorConfig *operatorv1.Console) string {
+	if operatorConfig.Spec.Providers.Statuspage != nil {
+		return operatorConfig.Spec.Providers.Statuspage.PageID
+	}
+	return ""
+}
+
+func DefaultConfigMap(
+	operatorConfig *operatorv1.Console,
+	consoleConfig *configv1.Console,
+	managedConfig *corev1.ConfigMap,
+	infrastructureConfig *configv1.Infrastructure,
+	rt *routev1.Route) (consoleConfigmap *corev1.ConfigMap, unsupportedConfigMerged bool, err error) {
+
+	b := &ConsoleYAMLConfigBuilder{}
+	defaultConfig, err := b.Host(rt.Spec.Host).
+		LogoutURL(defaultLogoutURL).
+		Brand(DEFAULT_BRAND).
+		DocURL(DEFAULT_DOC_URL).
+		APIServerURL(getApiUrl(infrastructureConfig)).
+		ConfigBytes()
+
 	extractedManagedConfig := extractYAML(managedConfig)
 
-	// Derive a user defined config using operator config
-	logoutRedirect := consoleConfig.Spec.Authentication.LogoutRedirect
-	brand := operatorConfig.Spec.Customization.Brand
-	docURL := operatorConfig.Spec.Customization.DocumentationBaseURL
-	providers := operatorConfig.Spec.Providers
-	userDefinedConfig := NewYamlConfig(host, logoutRedirect, brand, docURL, apiServerURL, providers)
+	b2 := &ConsoleYAMLConfigBuilder{}
+	userDefinedConfig, err := b2.Host(rt.Spec.Host).
+		LogoutURL(consoleConfig.Spec.Authentication.LogoutRedirect).
+		Brand(operatorConfig.Spec.Customization.Brand).
+		DocURL(operatorConfig.Spec.Customization.DocumentationBaseURL).
+		APIServerURL(getApiUrl(infrastructureConfig)).
+		StatusPageID(statusPageId(operatorConfig)).
```

Comment:
not sure if we shouldnt have public function per config field, eg.:
- Customization (instead of Brand and DocURL)
- Providers (instead of StatusPageID)
- ...


each of this suggested function can have subfunctions that can be tested individually but also as a whole...
...just thinking out loud, so we find a proper way how to deal with this fields :)

Reply:
I see now that you created a new struct for the `ConsoleYAMLConfigBuilder` ... in that case I think it makes sense to have field for each property .... thinking about the downsides though ...

Reply:
Yeah I want to eliminate this interim `yaml.MapSlice{ }` stuff.  Its a pain.  When we can get [config.go](https://github.com/openshift/console/blob/master/cmd/bridge/config.go) from the console:

```golang
type Config struct {
	APIVersion    string `yaml:"apiVersion"`   // yay! yaml tags! wooo
	Kind          string `yaml:"kind"`                  // yay! yaml tags! wooo
	ServingInfo   `yaml:"servingInfo"`            // yay! yaml tags! wooo
        // stuff....
	Providers     `yaml:"providers"`                // yay! yaml tags! wooo
}
// more stuff....
```
We can just take advantage of the `yaml` struct tags like for the Marshalling.

Reply:
Maybe I should just copy over that config in this PR.

Reply:
Yeah I think we want to have an interface that takes `1:1` items as they are set.... and behind the scenes will generate the complicated structure & stamp out the YAML if asked.

Reply:
Update: Copied over the Config into `subresource/consoleserver/config.go`.

Was able to eliminate that extra YAML library as we can take advantage of the `yaml` tags now.

Reply:
yeah, but shouldn't we rather import the pkg from the `console` repo ?